src/rars/venus/editors/jeditsyntax/JEditTextArea.java

```diff
@@ -141,15 +141,17 @@ public JEditTextArea(TextAreaDefaults defaults, JComponent lineNumbers) {
         caretTimer.setDelay(caretBlinkRate);
 
         // Intercept keystrokes before focus manager gets them.  If in editing window,
-        // pass TAB keystrokes on to the key processor instead of letting focus
+        // pass (SHIFT) TAB keystrokes on to the key processor instead of letting focus
         // manager use them for focus traversal.
         // One can also accomplish this using: setFocusTraversalKeysEnabled(false);
         // but that seems heavy-handed.
         // DPS 12May2010
         KeyboardFocusManager.getCurrentKeyboardFocusManager().addKeyEventDispatcher(
                 new KeyEventDispatcher() {
                     public boolean dispatchKeyEvent(KeyEvent e) {
-                        if (JEditTextArea.this.isFocusOwner() && e.getKeyCode() == KeyEvent.VK_TAB && e.getModifiers() == 0) {
+                        int modifiers = e.getModifiers();
+                        if (JEditTextArea.this.isFocusOwner() && e.getKeyCode() == KeyEvent.VK_TAB
```

Comment:
This seems to be overcomplicating the matter a bit, but if simplified to `JEditTextArea.this.isFocusOwner() && e.getKeyCode() == KeyEvent.VK_TAB` CTRL + TAB results in a stack overflow for some reason.